recipe-server/normandy/recipes/api/fields.py

```diff
@@ -42,3 +42,11 @@ def get_url(self, obj, view_name, request, format):
             url = reverse(view_name, kwargs=kwargs, request=request, format=format)
             assert url[0] == '/'
             return settings.CDN_URL + url[1:]
+
+    def to_representation(self, value):
+        # HyperlinkedIdentityField demands a request when creating
+        # urls, but one is not always available. Instead of erroring
+        # out hard, fake it.
```

Comment:
I think I'm a little confused then; I would think that `reverse`, when not given a `request` parameter, returns a non-absolute URL. If this is used in `canonical_json` during signing, wouldn't the data being signed have a domain-less URL and thus not match up with the data we server the user, which has a domain as part of the URL?